### PR TITLE
[ci/release] Default Python version to 3.8 for release tests

### PR DESCRIPTION
**Pull request by krfricke**
## Why are these changes needed?

After moving our CI to Python 3.8, we should also move our release tests to 3.8 per default.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by aslonnie — COMMENTED** [review]
I am not sure about the impact of this. maybe leave it for @can-anyscale to review tomorrow?

does this need to be cherrypicked into ray 2.6 release branch?

File: release/release_tests.yaml [hunk 1]
```diff
@@ -87,7 +87,7 @@
 
   frequency: nightly
   team: ml
-  python: "3.8"
+
```
Comment (aslonnie) [inline release/release_tests.yaml:90]: nit: just remove the empty line?

**Review by aslonnie — COMMENTED** [review]
are there release tests that are still using 3.7 today? is it possible to make this change a noop by adding `python: "3.7"` to ones that were still using default today?

**Comment by krfricke** [comment]
> are there release tests that are still using 3.7 today? is it possible to make this change a noop by adding `python: "3.7"` to ones that were still using default today?

Good point, I've added `python: "3.7"` to the ones that didn't have any defaults. We can upgrade these in follow-ups.

```
rllib_learning_tests_a2c_tf.aws
rllib_learning_tests_a2c_tf.gce
rllib_learning_tests_a2c_torch.aws
rllib_learning_tests_a2c_torch.gce
rllib_learning_tests_apex_tf.aws
rllib_learning_tests_apex_tf.gce
rllib_learning_tests_apex_torch.aws
rllib_learning_tests_apex_torch.gce
rllib_learning_tests_appo_tf.aws
rllib_learning_tests_appo_tf.gce
rllib_learning_tests_appo_torch.aws
rllib_learning_tests_appo_torch.gce
rllib_learning_tests_appo_old_stack_tf.aws
rllib_learning_tests_appo_old_stack_tf.gce
rllib_learning_tests_appo_old_stack_torch.aws
rllib_learning_tests_appo_old_stack_torch.gce
rllib_learning_tests_ddpg_torch.aws
rllib_learning_tests_ddpg_torch.gce
rllib_learning_tests_dqn_tf.aws
rllib_learning_tests_dqn_tf.gce
rllib_learning_tests_dqn_torch.aws
rllib_learning_tests_dqn_torch.gce
rllib_learning_tests_impala_tf.aws
rllib_learning_tests_impala_tf.gce
rllib_learning_tests_impala_torch.aws
rllib_learning_tests_impala_torch.gce
rllib_learning_tests_marwil_tf.aws
rllib_learning_tests_marwil_tf.gce
rllib_learning_tests_marwil_torch.aws
rllib_learning_tests_marwil_torch.gce
rllib_learning_tests_ppo_tf.aws
rllib_learning_tests_ppo_tf.gce
rllib_learning_tests_ppo_torch.aws
rllib_learning_tests_ppo_torch.gce
rllib_learning_tests_ppo_old_stack_tf.aws
rllib_learning_tests_ppo_old_stack_tf.gce
rllib_learning_tests_ppo_old_stack_torch.aws
rllib_learning_tests_ppo_old_stack_torch.gce
rllib_learning_tests_sac_tf.aws
rllib_learning_tests_sac_tf.gce
rllib_learning_tests_slateq_torch.aws
rllib_learning_tests_slateq_torch.gce
rllib_learning_tests_td3_torch.aws
rllib_learning_tests_td3_torch.gce
rllib_multi_gpu_learning_tests.aws
rllib_multi_gpu_learning_tests.gce
rllib_multi_gpu_with_lstm_learning_tests.aws
rllib_multi_gpu_with_lstm_learning_tests.gce
scheduling_test_many_0s_tasks_many_nodes.aws
scheduling_test_many_0s_tasks_many_nodes.gce
dataset_shuffle_push_based_random_shuffle_100tb.aws
dataset_shuffle_push_based_random_shuffle_100tb.gce
k8s_serve_ha_test
gcp_cluster_launcher_minimal
gcp_cluster_launcher_full
```

**Comment by krfricke** [comment]
Thanks @can-anyscale, I reverted the settings for the rllib tests, ptal!

**Review by can-anyscale — APPROVED** [review]

**Review by aslonnie — APPROVED** [review]

**Pull request merged**